[Chua Chee Hing Daryl] iP

rylzxc opened the pull request:
# Duke Header

* This is a bullet point
* Another point

1. This is a numbered point
2. Another point

A fenced code block with syntax highlighting looks something like this:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
A tasklist:
- [x] Managing tasks
- [ ] Do ~~some~~ lots of [code cleanup](https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley)
- [ ] Add **delight** to the experience when all tasks are complete :tada:

I am not a *quote*
> But I am a quote

Commit:
Add Gradle support

rylzxc committed:
Complete Level 1

rylzxc committed:
Complete Level 2

rylzxc committed:
Complete Level 3

rylzxc committed:
Complete Level-4

rylzxc committed:
Complete Level 5

rylzxc committed:
Complete Level 6

rylzxc committed:
Complete Automated Text UI Testing

rylzxc committed:
Some code refactoring

rylzxc committed:
Complete Level 7 and Level 8

rylzxc committed:
Complete Level-8

rylzxc committed:
Refactor code to abstract classes

rylzxc committed:
Improve exception handling

rylzxc committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

rylzxc committed:
Add Gradle Support

rylzxc committed:
Create JAR file

rylzxc committed:
Add JUnit Testcases

rylzxc committed:
Create A-JavaDoc branch

rylzxc committed:
Add some JavaDocs comments

rylzxc committed:
Complete A-JavaDoc

rylzxc committed:
Refactor code to comply with coding standards

rylzxc committed:
Complete Level 9

rylzxc committed:
Resolve merge conflict in branch 'branch-A-CodingStandard'

rylzxc committed:
Merge branch 'branch-A-JavaDoc'

rylzxc committed:
Add basic GUI

rylzxc committed:
Use Varargs

rylzxc committed:
Refactor GUI code

rylzxc committed:
Implement argument-by-argument GUI

reneeyeow02 commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,39 @@
+        taskList.addDeadlineTask(description, deadline);
+        storage.saveList(taskList);
+    }
+}
```
I think there should be a new line at the end of the file

reneeyeow02 commented on src/main/java/duke/common/Messages.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * List of messages which can be shown to user.
+ */
+public class Messages {
```
Would it be better to use enums for this instead of making everything public static final Strings?

reneeyeow02 commented on src/main/java/duke/exceptions/InvalidInputException.java:
```diff
@@ -0,0 +1,13 @@
+package duke.exceptions;
+
+/** Represents an exception for giving invalid input.*/
+public class InvalidInputException extends Exception {
```
Maybe should extend from IllegalArgumentException instead since this is a runtime exception

reneeyeow02 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * Constructor for UI.
+     */
+//    public Ui() {
```
Can remove commented out stuff

Kok-je replied:
If this is not needed, could consider removing it

Kok-je commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,57 @@
+     * @return the user command
+     */
+    public String getUserCommand() {
+        out.print("What's next: ");
```
Perhaps adding \t tab indentations or dividers to the displayed message can help with the readability.

Kok-je commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+        System.out.printf(MESSAGE_TASK_MARKED, taskNumber);
+    }
+
+    public void findTask(String taskSubstring) {
```
really like the specification of the string parameter as taskSubstring, helps me to identify what string is being passed into the findTask function!

Kok-je commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+        }
+
+        if (sb.toString().isEmpty()) {
+            System.out.println(MESSAGE_NO_TASK_FOUND);
```
Perhaps you can consider abstraction the sout message to an exception error?

Kok-je commented on src/main/java/duke/task/DeadlineTask.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+/**
```
Should include an empty line here

Kok-je commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,148 @@
+                throw new InvalidInputException(Messages.MESSAGE_INVALID_TASK_NUMBER);
+            }
+            return new MarkCommand(Integer.parseInt(taskNumber));
+        } catch (Exception e) {
```
Can consider encapsulating invalid command as a Duke exception here

Kok-je reviewed (commented):
Great job Daryl! Overall, the code was very easy to read because the indentation and whitespace rules are observed very carefully. Just made a few minor comments on the code that you can look through!

reneeyeow02 reviewed (commented):
Code was overall pretty readable, everything is packaged neatly

rylzxc committed:
Merge branch 'branch-A-Varargs'

rylzxc committed:
Add asserions to methods in StorageFile and Ui

Asserted that storage file has to be created if it does not exist, else it would be a bug in the program.

Asserted that returnMessages() in Ui has to return messages to follow the meaning of its name.

Might need to handle these bugs early with exception handling.

rylzxc committed:
Add assertion in TaskList

Assert that task has successfully been removed from the list.

Otherwise the taskCount will not tally with the total number of tasks.

rylzxc committed:
Improve code quality

Removed unused methods, rewrote star imports and removed commented out code.

Unused methods and commented out code pertained to CLI UI as compared to JavaFX UI therefore inapplicable.

rylzxc committed:
Add lexiographical sort functionality

Users may want to order their tasks according to alphabet

rylzxc committed:
Test PR

rylzxc committed:
Merge pull request #1 from rylzxc/PR-branch

Test PR

rylzxc committed:
Improve help message

rylzxc committed:
Improve help message

rylzxc committed:
Update README

rylzxc committed:
Add picture of Ui

rylzxc committed:
Update README

rylzxc committed:
Update testcases
